samples/AudioPlayerSample/ViewModels/AudioRecorderPageViewModel.cs

```diff
@@ -4,36 +4,117 @@ namespace AudioPlayerSample.ViewModels;
 
 public class AudioRecorderPageViewModel : BaseViewModel
 {
-	readonly IAudioManager audioManager;
-	IAudioRecorder? audioRecorder;
+   ContentPage page;
 
-	public AudioRecorderPageViewModel(
-		IAudioManager audioManager)
-	{
-		StartCommand = new Command(Start);
+   readonly IAudioManager audioManager;
+   IAudioRecorder audioRecorder;
+   IAudioPlayer audioPlayer;
+   IAudioSource audioSource = null;
 
-		this.audioManager = audioManager;
-	}
+   string recordButtonText = "Record";
+   public string RecordButtonText
+   {
+      get => recordButtonText;
+      set
+      {
+         recordButtonText = value;
+         NotifyPropertyChanged();
+      }
+   }
 
-	public Command StartCommand { get; set; }
+   Color recordButtonColor = Colors.Blue;
+   public Color RecordButtonColor
+   {
+      get => recordButtonColor;
+      set
+      {
+         recordButtonColor = value;
+         NotifyPropertyChanged();
+      }
+   }
```

Comment:
I would prefer to stick to firmer MVVM concepts and manage this part in the view/page. We can swap colors/text based on a state or boolean property to know whether the view model is currently recording.